imagebuildah/stage_executor.go

```diff
@@ -1168,6 +1194,24 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 			if err != nil {
 				return "", nil, fmt.Errorf("error checking if cached image exists from a previous build: %w", err)
 			}
+			// All the best effort to find cache on localstorage have failed try pulling
+			// cache from remote repo if `--cache-from` was configured.
+			if cacheID == "" && s.executor.cacheFrom != "" {
+				src := fmt.Sprintf("%s:%s", s.executor.cacheFrom, cacheKey)
```

Comment:
Looking at https://github.com/containers/buildah/pull/4118#discussion_r924478405 and https://github.com/containers/buildah/pull/4118#issuecomment-1189033450 … let me take a rather different tack:

- The c/image image names have no _common_ semantics apart from the initial _transport_`:` part. There is no expectation that one can just add a `:`_cacheKey_ to user input (let alone to a valid c/image image name), and have that be a valid name.
- What would we want to use this for, apart from `docker://`, anyway? I suppose `dir:` or `oci:` is plausible … for some kind of inter-storage cache sharing???? It’s not obvious to me that that’s better than a registry with compressed data that can just be reused on the registry via blob mounting.
- Given the Redis references… I think being explicit would be nice anyway.

So, how about a `cache{From,To}{Docker,Registry}Repo reference.Named`, parsed at the CLI via `reference.ParseNormalizedNamed()`+ an enforcement of `reference.IsNameOnly`? Then, the actual image name would be determined via `docker.NewReference(reference.WithTag(cache…Repo, cacheKey))`.

For other storage types, we would just add more option keys.

@nalind @rhatdan WDYT?

Reply:
@mtrmac I like the idea of doing parsing and forming this at CLI will open room for lot more possiblities

Reply:
@mtrmac One doubt would it makes sense to do generation of `reference.Named` at backend (imagebuildah) and just pass `string` from frontend API my intention is to make frontend API easy to consume and let imagebuildah take care of everything ?  But just a suggestion and no issues from my side with doing it at frontend/CLI.

Reply:
It's not a prohibitively complicated requirement (`buildah.Push()` returns a `reference.Canonical`) for an optional setting, and I think the loss of ambiguity would be well worth it.

Reply:
Cool I agree, I'll make this change in the next push.

Reply:
implemented on CLI